### ISSUE TITLE
"Failed to find network for networkMode managed"

### ISSUE THREAD

**Issue by joshuacox** [rancher/rancher#7108]
**Rancher Version:**  v1.2.1

**Docker Version:**

```
rancher@rancher:~$  docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.3
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.24-rancher
Operating System: RancherOS v0.7.1
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 2.011 GiB
Name: rancher
ID: H3A2:IATS:DWLG:MLLB:W7JZ:VEBG:6QU4:HLDJ:47ZL:IEWA:UOX2:Y6QF
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

```

**OS and where are the hosts located? (cloud, bare metal, etc):**

rancherOS 071 in KVM virtual machine

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

3  rancherOS 071 1 server node + 2 agent nodes

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle

**Steps to Reproduce:**
install fresh rancher 071 install server container, register both agent nodes and install anything you like so far as I can tell, but gitlab for example

**Results:** 

```
1 Sidekick (Failed to allocate instance [container:1i274]: Bad instance [container:1i274] in state [removed]: Failed to find network for networkMode managed
```

**Expected:**

working networkMode managed

**more info**

```
rancher@rancher:~$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                              NAMES
86317845af38        rancher/server      "/usr/bin/entry /usr/"   2 hours ago         Up 36 minutes       3306/tcp, 0.0.0.0:8080->8080/tcp   jovial_pike
rancher@rancher:~$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
rancher/server      latest              5a9f3387a325        36 hours ago        931.6 MB
```

I'll mention https://github.com/rancher/rancher/issues/6573 but this install was 1.2.1 from the beginning

**Comment by deniseschannon** [comment]
Did all the infrastructure stacks become green before you launched anything?

**Comment by aemneina** [comment]
@joshuacox what kind of networking are you using for the VM's (bridged, vlans, iptables...)?

**Comment by joshuacox** [comment]
@aemneina  the VMs are bridged at the KVM host level, effectively getting peer addresses on the 10.x.x.x network

**Comment by joshuacox** [comment]
@deniseschannon  sorry I erroneously put 'cattle', it's all KVM but I installed from the rancherOS ISO as if they were 'bare-metal'

**Comment by joshuacox** [comment]
here is an API dump of one of the active agents:

```
{    "id": "1h3",
    "type": "host",
    "links": {
        "self": "…/v2-beta/projects/1a5/hosts/1h3",
        "account": "…/v2-beta/projects/1a5/hosts/1h3/account",
        "clusters": "…/v2-beta/projects/1a5/hosts/1h3/clusters",
        "containerEvents": "…/v2-beta/projects/1a5/hosts/1h3/containerevents",
        "healthcheckInstanceHostMaps": "…/v2-beta/projects/1a5/hosts/1h3/healthcheckinstancehostmaps",
        "hostLabels": "…/v2-beta/projects/1a5/hosts/1h3/hostlabels",
        "hosts": "…/v2-beta/projects/1a5/hosts/1h3/hosts",
        "instances": "…/v2-beta/projects/1a5/hosts/1h3/instances",
        "ipAddresses": "…/v2-beta/projects/1a5/hosts/1h3/ipaddresses",
        "physicalHost": "…/v2-beta/projects/1a5/hosts/1h3/physicalhost",
        "serviceEvents": "…/v2-beta/projects/1a5/hosts/1h3/serviceevents",
        "storagePools": "…/v2-beta/projects/1a5/hosts/1h3/storagepools",
        "volumes": "…/v2-beta/projects/1a5/hosts/1h3/volumes",
        "stats": "…/v2-beta/projects/1a5/hosts/1h3/stats",
        "hostStats": "…/v2-beta/projects/1a5/hosts/1h3/hoststats",
        "containerStats": "…/v2-beta/projects/1a5/hosts/1h3/containerstats"
    },
    "actions": {
        "dockersocket": "…/v2-beta/projects/1a5/hosts/1h3/?action=dockersocket",
        "update": "…/v2-beta/projects/1a5/hosts/1h3/?action=update",
        "deactivate": "…/v2-beta/projects/1a5/hosts/1h3/?action=deactivate"
    },
    "baseType": "host",
    "name": null,
    "state": "active",
    "accountId": "1a5",
    "agentIpAddress": "10.x.x.97",
    "agentState": "active",
    "authCertificateAuthority": null,
    "authKey": null,
    "azureConfig": null,
    "computeTotal": 1000000,
    "created": "2016-12-17T19:33:20Z",
    "createdTS": 1482003200000,
    "description": null,
    "digitaloceanConfig": null,
    "dockerVersion": null,
    "driver": null,
    "engineEnv": null,
    "engineInsecureRegistry": null,
    "engineInstallUrl": null,
    "engineLabel": null,
    "engineOpt": null,
    "engineRegistryMirror": null,
    "engineStorageDriver": null,
    "hostname": "rancher",
    "info": {
        "iopsInfo": { },
        "memoryInfo": {
            "active": 379,
            "buffers": 86,
            "cached": 841,
            "inactive": 632,
            "memAvailable": 1534,
            "memFree": 887,
            "memTotal": 2001,
            "swapCached": 0,
            "swapfree": 0,
            "swaptotal": 0
        },
        "osInfo": {
            "dockerVersion": "Docker version 1.12.3, build 6b644ec",
            "kernelVersion": "4.4.24",
            "operatingSystem": "RancherOS v0.7.1"
        },
        "diskInfo": {
            "dockerStorageDriver": "overlay",
            "dockerStorageDriverStatus": { },
            "fileSystems": {
                "/dev/vda1": {
                    "capacity": 31548
                }
            },
            "mountPoints": {
                "/dev/vda1": {
                    "free": 28712,
                    "percentage": 3.582575577479298,
                    "total": 31548,
                    "used": 1130
                }
            }
        },
        "cpuInfo": {
            "count": 1,
            "cpuCoresPercentages": [
                1.0000000000218279
            ],
            "loadAvg": [ 3 items
                "0",
                "0.01",
                "0"
            ],
            "mhz": 1497.191,
            "modelName": "AMD Opteron 23xx (Gen 3 Class Opteron)"
        }
    },
    "instanceIds": [ 3 items
        "1i38",
        "1i41",
        "1i160"
    ],
    "kind": "docker",
    "labels": {
        "io.rancher.host.agent_image": "rancher/agent:v1.1.1",
        "io.rancher.host.docker_version": "1.12",
        "io.rancher.host.linux_kernel_version": "4.4",
        "space": "oltorf",
        "activeDeploy": "true",
        "os": "rancher"
    },
    "localStorageMb": 30113656,
    "memory": 2098200576,
    "milliCpu": 1000,
    "physicalHostId": "1ph3",
    "publicEndpoints": null,
    "removed": null,
    "transitioning": "no",
    "transitioningMessage": null,
    "transitioningProgress": null,
    "uuid": "xxxxxxxxxxxxxxxxxxxxxxxxx"

}
```

**Comment by joshuacox** [comment]
I notice the default environment was using a template of 'none', so I removed the hosts under infrastructure, created a new environment, using the 'cattle' template.  And have added the hosts, and they began to populate wtih the usual networking infrastructure containers.  I started with a gitlab container again and it seems to have gotten past the networking issue, all seems to be good I'm going to do a little more testing before I close the issue, is there something wrong with the Default environment?  I did another installation from ISO and then installed rancherserver and again the Default environment had a template of 'none', under the manage environments interface inside of rancher.

**Comment by mjaverto** [comment]
Just to comment for anyone else with this issue that doesn't have resolution from above. For background, we've been running rancher since July '16.

1. I installed fresh rancher 1.2.x due to migration issues from previous versions.
1. Created new environments for staging and production
1. (stupidly) removed the new stacks rancher > 1.2.x has that I thought were extra that I didn't see in past version.

I ended up creating an entirely new DB before realizing rancher > 1.2 has some infrastructure stacks that are vital for managed networks.
